### ISSUE TITLE
Allow custom validation messages

### ISSUE THREAD

**Issue by tflori** [getopt-php/getopt-php#118]
Currently the message is always `'Option \'%s\' has an invalid value'`. It would be helpful to have more descriptive messages. You can have this already in custom validation functions, but it's not possible for php functions. The proposal was to pass a second variable for validation.

Example for custom validation:

```php
$option = Option::create('u', 'user')
  ->setValidation(function ($value) {
    if (!DI::em()->fetch('User', $value)) {
      throw new Invalid(sprintf('User %s not found', $value));
    }
    return true;
  });
```

Example using php functions (new feature):

```php
$option = Option::create('a', 'alpha')
  ->setValidation('is_numeric', 'Alpha has to be numeric');
```

**Comment by tflori** [comment]
I was just thinking about more functionality for validation of values but we support all these cases already.

For example multiple validations:

```php
$alpha = Option::create('a', 'alpha')
  ->setValidation(function ($value) {
    if (!is_numeric($value)) {
      throw new Invalid('Alpha has to be numeric');
    }
    $alpha = float($value);
    if ($alpha < 0 or $alpha > 1) {
      throw new Invalid('Alpha has to be between 0 and 1');
    }
    return true;
  });
```

If you want the validations to be more detailed during `GetOpt::process()` you can use 3rd part libraries:

```php
// for example tflori/verja
$field = new Verja\Field(['integer', 'between:0:255']);
$alpha = Option::create('a', 'alpha')
  ->setValidation([$field, 'validate']);
```

What we need is to be able to create the validation message with a callable:

```php
// for example tflori/verja
$field = new Verja\Field(['integer', 'between:0:255']);
$alpha = Option::create('a', 'alpha')
  ->setValidation([$field, 'validate'], function () use ($field) {
    $message = 'Option alpha has an invalid value:' . PHP_EOL;
    foreach ($field->getErrors() as $error) {
      $message .= '  ' . $error->message . PHP_EOL;
    }
    return $message;
  });
```

**Comment by tflori** [comment]
Anyway I would recommend to use [tflori/verja](https://github.com/tflori/verja) different:

```php
$getOpt = new GetOpt\GetOpt([
  GetOpt\Option::create('a', 'alpha'),
  GetOpt\Option::create('r', 'red'),
  GetOpt\Option::create('g', 'green'),
  // ..
]);
$getOpt->process();

$gate = new Verja\Gate();
$gate->accepts([
  'alpha' => ['numeric', 'between:0:1'],
  'red' => ['integer', 'between:0:255'],
  'green' => ['integer', 'between:0:255'],
  // ...
]);
// this throws - you can use validate if you want to translate errors: read the docs...
$data = $gate->getData($getOpt->getOptions());
```

**Comment by tflori** [comment]
solved with #120
